examples/private_ai_studio/main.tf

```diff
@@ -0,0 +1,90 @@
+terraform {
+  required_version = "~> 1.5"
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~> 3.115.0"
+    }
+  }
+}
+
+provider "azurerm" {
+  features {
+    key_vault {
+      purge_soft_delete_on_destroy = false
+    }
+    resource_group {
+      prevent_deletion_if_contains_resources = false
+    }
+  }
+}
+
+# This ensures we have unique CAF compliant names for our resources.
+module "naming" {
+  source  = "Azure/naming/azurerm"
+  version = "~> 0.3"
+}
+
+# This is required for resource modules
+resource "azurerm_resource_group" "this" {
+  location = var.location
+  name     = module.naming.resource_group.name_unique
+}
+
+locals {
+  name = module.naming.machine_learning_workspace.name_unique
+}
+
+# This is the module call
+# Do not specify location here due to the randomization above.
+# Leaving location as `null` will cause the module to use the resource group location
+# with a data source.
+module "aihub" {
+  source = "../../"
+  # source             = "Azure/avm-<res/ptn>-<name>/azurerm"
+  # ...
+  location            = azurerm_resource_group.this.location
+  name                = local.name
+  resource_group_name = azurerm_resource_group.this.name
+  is_private          = true
+  kind                = "Hub"
+  workspace_managed_network = {
+    isolation_mode = "AllowInternetOutbound"
+    spark_ready    = true
+  }
+
+  container_registry = {
+    create_new     = true
+    zone_redundant = false
+  }
+
+  key_vault = {
+    create_new = true
+    network_acls = {
+      bypass         = "AzureServices"
+      default_action = "Deny"
+    }
+  }
+
+  storage_account = {
+    create_new = true
+    network_rules = {
+      bypass         = ["Logging", "Metrics", "AzureServices"]
+      default_action = "Deny"
```

Comment:
Same this should probably be automatically done but not for this PR but maybe wrth for us to create an issue to track this.